data-reconciliation-app/test/e2e-test/src/index.ts

```diff
@@ -0,0 +1,196 @@
+        this.assertReportField(member.name, reportItem, 'count_of_unique_values', 2);
+        this.assertReportField(member.name, reportItem, 'members_in_agreement', 1);
+
+        // member0
+        recordId = '984500F5BD5BE5767C51';
+        console.log(`\nChecking fields for ${member.name} and id ${recordId} (In Consensus)\n`);
+        reportItem = await Api.reportById(this.demoProps, member, recordId);
+        this.assertReportField(member.name, reportItem, 'group_status', 'IN_CONSENSUS');
+        this.assertReportField(member.name, reportItem, 'majority_minority', 'Majority');
+        this.assertReportField(member.name, reportItem, 'count_of_unique_values', 1);
+        this.assertReportField(member.name, reportItem, 'members_in_agreement', 3);
+
+        member = this.members[1];
+        recordId = '984500E1B2CA1D4EKG67';
+        console.log(`\nChecking fields for ${member.name} and id ${recordId} (Lack of Consensus)\n`);
+        reportItem = await Api.reportById(this.demoProps, member, recordId);
+        this.assertReportField(member.name, reportItem, 'lei', recordId);
+        this.assertReportField(member.name, reportItem, 'nace', 'A01.1');
+        this.assertReportField(member.name, reportItem, 'group_status', 'LACK_OF_CONSENSUS');
+
+        // member1
+        recordId = '984500F5BD5BE5767C51';
+        console.log(`\nChecking fields for ${member.name} and id ${recordId} (In Consensus)\n`);
+        reportItem = await Api.reportById(this.demoProps, member, recordId);
+        this.assertReportField(member.name, reportItem, 'group_status', 'IN_CONSENSUS');
+
+        this.printTestSectionHeader('🎉 All Tests Passed...');
+    }
+
+
+    private static assertReportField(memberName: string, reportItem: ReportItem, fieldName: string, expectedValue: string | number) {
+        const currentValue = reportItem[fieldName];
+        if (currentValue == expectedValue) {
+            console.log(`✅ [PASS] - Assert ${memberName}::${reportItem.lei}.${fieldName} == ${expectedValue}`);
+        } else {
+            throw new Error(`🛑 [TEST FAILURE]: Unexpected ${fieldName} for ${memberName}::${reportItem.lei} - Current: ${currentValue}. Expected: ${expectedValue}`);
+        }
+    }
+
+    private static createMember(memberId: string): DemoMemberProps {
+        return {
+            id: memberId,
+            name: `Member ${memberId}`,
+            data: this.memberDataMap.get(memberId),
+            httpsAgent: this.createHttpsAgent(memberId),
+        };
+    }
+
+    private static createHttpsAgent(memberId: string): https.Agent {
+        return new https.Agent({
+            cert: fs.readFileSync(`${certificateStorePath}/member${memberId}_cert.pem`),
+            key: fs.readFileSync(`${certificateStorePath}/member${memberId}_privk.pem`),
+            ca: fs.readFileSync(`${certificateStorePath}/service_cert.pem`),
+        })
+    }
+
+    private static printTestSectionHeader(title: string) {
+        console.log('\n\n');
```

Comment:
Does this need so many newlines around it? Makes my console hard to scroll!

Reply:
reduced